content/knowledge-base/migrate-to-github-app.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Migrating from GitHub OAuth to GitHub App
+weight: 1
+---
+
+On April 19, 2021, Codemagic will deprecate the GitHub OAuth integration due to GitHub ending support for OAuth App integrations, you can read more about it [here](https://developer.github.com/changes/2020-02-14-deprecating-oauth-app-endpoint/). We are asking everyone currently logging in and accessing their GitHub repositories in Codemagic via the GitHub OAuth integration to switch to the GitHub App to continue building their GitHub repositories after the deprecation date.
+
+## Who are affected?
+
+If you signed up or log in to Codemagic via GitHub (All repositories, OAuth), then this means we are using the OAuth integration to access the repositories on your GitHub account. In the Integrations section in your user or team settings, you can see the GitHub OAuth integration enabled.
+
+GitHub repositories accessed via OAuth are marked with a yellow warning icon in your list of applications. A warning is also displayed in the team settings if any of the apps shared in a team is accessed via the OAuth integration. Once the OAuth integration is removed, such repositories will no longer be accessible unless you switch to GitHub App.
+
+Deprecating OAuth does not affect GitHub repositories that are added over SSH or HTTPS.
+
+## GitHub OAuth vs GitHub App
+
+The key difference between OAuth and GitHub App integrations is the scope of permissions. While OAuth requires you to grant Codemagic **write access** to all your repositories, the GitHub App integration requests only **read access** to your code and you can select which repositories you share with Codemagic.
+
+In addition, the GitHub app integration will also make it possible to use [GitHub Checks](../building/github-checks).
+
+### Deprecting GitHub releases in Flutter workflow editor
+
+Without write access to the repository, Codemagic will no longer be able to push GitHub releases without additional configuration. Therefore, we will be removing the GitHub releases section in Flutter workflow editor. You can continue to
+publish GitHub releases by setting up a personal access token and using a custom script in the ... step.
```

Comment:
```suggestion
publish GitHub releases by setting up a personal access token and using a custom script in the Pre-publish step.

```bash
#!/usr/bin/env zsh

# Make sure that you have exported your GitHub personal access token as
# environment variable `GITHUB_TOKEN`. GitHub CLI uses token from this 
# environment variable for authentication.

# Publish only for tag builds
if [ -z ${FCI_TAG} ]; then
   echo "Not a tag build will not publish GitHub release"
   exit 0
fi

# See more options about `gh release create` usage from GitHub CLI
# official docs at https://cli.github.com/manual/gh_release_create

gh release create "${FCI_TAG}" \
    --title "<Your Application Name> ${FCI_TAG}" \
    --notes-file changelog.md
    path/to/build-artifact.ipa \
    path/to/build-artifact.apk
```

# Note that you don't need to include title and changelog if you do not want to.
# Any number of artifacts can be included with the release.